tensordict/tensordict.py

```diff
@@ -2656,23 +2662,75 @@ def memmap_(self, prefix=None) -> TensorDictBase:
                     "memmap is not compatible with gradients, one of Tensors has requires_grad equals True"
                 )
             if isinstance(value, TensorDictBase):
-                self._tensordict[key] = value.memmap_()
+                if prefix is not None:
+                    # ensure subdirectory exists
+                    (prefix / key).mkdir(exist_ok=True)
+                    self._tensordict[key] = value.memmap_(prefix=prefix / key)
+                    torch.save(
+                        {"batch_size": value.batch_size, "device": value.device},
+                        prefix / key / "meta.pt",
+                    )
+                else:
+                    self._tensordict[key] = value.memmap_()
                 continue
-            self._tensordict[key] = MemmapTensor(value, prefix=prefix)
+            self._tensordict[key] = MemmapTensor(
+                value,
+                filename=str(prefix / f"{key}.memmap") if prefix is not None else None,
+            )
+            if prefix is not None:
+                torch.save(
```

Comment:
OOC could we save all the metadata at once?
load/save overhead is not too much a worry here though (I wouldn't suspect it to constitute a bottleneck in a typical training loop)

Reply:
I think in principle there's no particular reason we couldn't, it would just be a little bit more fiddly most likely. But agreed, there is some overhead from doing multiple reads / writes each time.

One thing though that is nice about the current approach though is that you can load a nested tensordict by pointing `TensorDict.load_memmap` at the nested directory. It can be oblivious to the parent directories.

Reply:
Yeah I agree let's leave it as is. 
Can we add to the doc that this is "supposed" to be slow, and should not be used to move things around during training?

Reply:
Added a note on slowness to docs and docstring